megamek/src/megamek/common/Board.java

```diff
@@ -897,15 +897,32 @@ public void load(InputStream is) {
         }
 
         // check data integrity
-        if ((nw > 1) || (nh > 1) || (di == (nw * nh))) {
+        if (isValid(nd) && ((nw > 1) || (nh > 1) || (di == (nw * nh)))) {
             newData(nw, nh, nd);
-        } else {
+        }else{
             System.err.println("board data invalid");
         }
 
     }
 
-    private int indexFor(String hexNum, int width, int row) {
+    public boolean isValid() {
+		//Search for black-listed hexes
+    	for(IHex hex: data) {
+    		if(hex == null)return false;
+    		if(!hex.isValid()) return false;	
+    	}
+		return true;
+	}
+    
+    private boolean isValid(IHex[] data) {
+		//Search for black-listed hexes
+    	for(IHex hex: data) {
+    		if(!hex.isValid()) return false;	
+    	}
+		return true;
+	}
+
```

Comment:
It seems like there's unnecessary code duplication here.  Why does `isValid(0)` not just call `isValid(1)`?  Why does `isValid(1)` not check for a null hex but `isValid(0)` does?